src/org/traccar/protocol/MiniFinderProtocolDecoder.java

```diff
@@ -94,49 +195,46 @@ protected Object decode(
-
-                int flags = parser.nextInt(16);
-
-                position.setValid(BitUtil.check(flags, 0));
-
-                if (BitUtil.check(flags, 2)) {
-                    position.set(Position.KEY_ALARM, Position.ALARM_FAULT);
-                }
-                if (BitUtil.check(flags, 6)) {
-                    position.set(Position.KEY_ALARM, Position.ALARM_SOS);
-                }
-                if (BitUtil.check(flags, 7)) {
-                    position.set(Position.KEY_ALARM, Position.ALARM_OVERSPEED);
-                }
-                if (BitUtil.check(flags, 8)) {
-                    position.set(Position.KEY_ALARM, Position.ALARM_FALL_DOWN);
-                }
-                if (BitUtil.check(flags, 9) || BitUtil.check(flags, 10) || BitUtil.check(flags, 11)) {
-                    position.set(Position.KEY_ALARM, Position.ALARM_GEOFENCE);
-                }
-                if (BitUtil.check(flags, 12)) {
-                    position.set(Position.KEY_ALARM, Position.ALARM_LOW_BATTERY);
-                }
-                if (BitUtil.check(flags, 15) || BitUtil.check(flags, 14)) {
-                    position.set(Position.KEY_ALARM, Position.ALARM_MOVEMENT);
-                }
-
-                position.set(Position.KEY_RSSI, BitUtil.between(flags, 16, 20));
-                position.set(Position.KEY_CHARGE, BitUtil.check(flags, 22));
-
-                position.setAltitude(parser.nextDouble());
-
-                position.set(Position.KEY_BATTERY, parser.next());
-                position.set(Position.KEY_SATELLITES, parser.next());
+            decodeFlags(position, parser.nextInt(16));
 
-            }
+            position.setAltitude(parser.nextDouble());
+
+            position.set(Position.KEY_BATTERY, parser.nextInt());
+
+            // Ignoring last three fields, unknown
+            position.set("unknown", parser.next() + "," + parser.next() + "," + parser.next());
+
+            position.set(Position.KEY_TYPE, sentence.substring(1, 2));
 
             return position;
+        }
 
+        if (sentence.startsWith("!A,")) {
+            Parser parser = new Parser(PATTERN_A, sentence);
+            if (!parser.matches()) {
+                return null;
+            }
+
+            Position position = new Position();
+            position.setProtocol(getProtocolName());
+            position.setDeviceId(deviceSession.getDeviceId());
+
+            DateBuilder dateBuilder = new DateBuilder()
+                    .setDateReverse(parser.nextInt(), parser.nextInt(), parser.nextInt())
+                    .setTime(parser.nextInt(), parser.nextInt(), parser.nextInt());
+            position.setTime(dateBuilder.getDate());
+
+            position.setLatitude(parser.nextDouble());
+            position.setLongitude(parser.nextDouble());
+
+            // Ignoring last three fields, unknown
+            position.set("unknown", parser.next() + "," + parser.next() + "," + parser.next());
```

Comment:
Do you agree that I can remove the !A and !C records from the protocol given that there is no documentation for such records?